### PR TITLE
Parity: TimeZone.local, TimeZone.timeZoneDataVersion

### PR DESCRIPTION
**Pull request by millenomi**
Add `__NSLocalTimeZone`; the name is chosen so as to be `NSCoding`-compatible with Darwin. Make sure `NSTimeZone.local` and `TimeZone.autoupdatingCurrent` return it.

Implement `.timeZoneDataVersion` the same way as it is in Darwin.

Re-enable the setter for `.abbreviationDictionary`. We'll fix its bugs rather than crashing.

Fixes https://bugs.swift.org/browse/SR-10348, https://bugs.swift.org/browse/SR-10349

### CODE REVIEW

**Comment by millenomi** [comment]
@swift-ci please test

File: Foundation/NSTimeZone.swift [hunk 1]
```diff
@@ -212,12 +211,17 @@ extension NSTimeZone {
...
-            // CFTimeZoneSetAbbreviationDictionary(newValue._cfObject)
-            NSUnimplemented()
+            CFTimeZoneSetAbbreviationDictionary(newValue._cfObject)
         }
     }
 
-    open class var timeZoneDataVersion: String { NSUnimplemented() }
+    open class var timeZoneDataVersion: String {
+        #if os(Windows)
+            return "" // We do not source timezone data from ICU. The empty string is what Darwin would return if ICU isn't able to give a time zone data version.
```
Comment (compnerd) [inline Foundation/NSTimeZone.swift]: Hmm, we should be able to do this on Windows I think.  We do have ICU available and will link against it.

Comment (millenomi) [reply Foundation/NSTimeZone.swift]: Ah, okay. Do you want me to just return the ICU version?

Comment (compnerd) [reply Foundation/NSTimeZone.swift]: Yes please!

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**